pkg/router/routeapihelpers/validation.go

```diff
@@ -375,3 +375,35 @@ func validateCertificatePEM(certPEM string, options *x509.VerifyOptions) ([]*x50
 
 	return certs, nil
 }
+
+// UpgradeRouteValidation performs an upgrade validation for
+// a route. This checks for issues that will cause failures in the next
+// OpenShift version.
+func UpgradeRouteValidation(route *routev1.Route) field.ErrorList {
```

Comment:
https://github.com/uber-go/nilaway

Reply:
I tried it out, I think it works, but I don't see any issues for the two main packages I'd be concerned with
```
[gspence@gspence-thinkpadp1gen3 router]$ nilaway -include-pkgs="controller" ./pkg/router/
[gspence@gspence-thinkpadp1gen3 router]$ nilaway -include-pkgs="routeapihelpers" ./pkg/router/
[gspence@gspence-thinkpadp1gen3 router]$
```